### PR TITLE
[autothrottle] path optimal throttle calculations

### LINKED ISSUE #81: [autothrottle] variable throttle rates by path
Autothrottle should build a map of src->dst paths and determine headroom, apply varying throttle rates by path. This would prevent a single broker hotspot from slowing down all other replications.

### PR DESCRIPTION
**Pull request by jamiealquiza**
# Overview
Currently, autothrottle mimics the out of the box reassign-partitions throttle parameter in that a single value is assigned to both the inbound and outbound throttle rates for all brokers participating in a reassignment. The single rate value is determined by observing metrics data at each check interval and making a calculation based on the most loaded broker.

Internally, Kafka doesn't actually require that the inbound and outbound rates for a given broker are the same, nor does it require rates among brokers to be the same value. This PR creates a graph of the reassignment and observes broker utilization according to role; a "follower" in a replication is receiving data and a "leader" is sending data. A per-path rate is calculated according to the configured instance type capacity limit, an observation of current network utilization (accounting for any previously set throttles), and a ratio (separately configurable for inbound vs outbound flow) of free network capacity that's eligible for replication use. For each path, the rate is updated if it's more than n (configurable) distance from any previously set rates.

tl;dr this ensure that every replication path individually runs as fast as possible while adaptively maintaining capacity for consumers. Closes #81.

# In Action

```
2020/02/27 22:28:12 Autothrottle Running
2020/02/27 22:28:13 Admin API: localhost:8080
2020/02/27 22:28:13 Topics with ongoing reassignments: [test0]
2020/02/27 22:28:13 Source brokers participating in replication: [1037 1039]
2020/02/27 22:28:13 Destination brokers participating in replication: [1033 1041]
2020/02/27 22:28:14 Replication throttle rate for broker 1037 [leader] (based on a 90% max free capacity utilization): 139.83MB/s
2020/02/27 22:28:14 Updated throttle on broker 1037 [leader]
2020/02/27 22:28:15 Replication throttle rate for broker 1039 [leader] (based on a 90% max free capacity utilization): 147.24MB/s
2020/02/27 22:28:15 Updated throttle on broker 1039 [leader]
2020/02/27 22:28:15 Replication throttle rate for broker 1041 [follower] (based on a 90% max free capacity utilization): 179.75MB/s
2020/02/27 22:28:15 Updated throttle on broker 1041 [follower]
2020/02/27 22:28:15 Replication throttle rate for broker 1033 [follower] (based on a 90% max free capacity utilization): 181.88MB/s
2020/02/27 22:28:15 Updated throttle on broker 1033 [follower]
2020/02/27 22:28:28 Topics with ongoing reassignments: [test0]
2020/02/27 22:28:28 Source brokers participating in replication: [1037 1039]
2020/02/27 22:28:28 Destination brokers participating in replication: [1033 1041]
2020/02/27 22:28:28 Replication throttle rate for broker 1039 [leader] (based on a 90% max free capacity utilization): 225.00MB/s
2020/02/27 22:28:28 Updated throttle on broker 1039 [leader]
2020/02/27 22:28:28 Replication throttle rate for broker 1041 [follower] (based on a 90% max free capacity utilization): 225.00MB/s
2020/02/27 22:28:28 Updated throttle on broker 1041 [follower]
2020/02/27 22:28:29 Replication throttle rate for broker 1033 [follower] (based on a 90% max free capacity utilization): 225.00MB/s
2020/02/27 22:28:29 Updated throttle on broker 1033 [follower]
2020/02/27 22:28:29 Replication throttle rate for broker 1037 [leader] (based on a 90% max free capacity utilization): 225.00MB/s
2020/02/27 22:28:29 Updated throttle on broker 1037 [leader]
2020/02/27 22:28:43 Topics with ongoing reassignments: [test0]
2020/02/27 22:28:43 Source brokers participating in replication: [1037]
2020/02/27 22:28:43 Destination brokers participating in replication: [1041]
2020/02/27 22:28:43 Replication throttle rate for broker 1037 [leader] (based on a 90% max free capacity utilization): 225.00MB/s
2020/02/27 22:28:43 Proposed throttle is within 0.00% of the previous throttle (below 10.00% threshold), skipping throttle update for broker 1037
2020/02/27 22:28:43 Replication throttle rate for broker 1041 [follower] (based on a 90% max free capacity utilization): 225.00MB/s
2020/02/27 22:28:43 Proposed throttle is within 0.00% of the previous throttle (below 10.00% threshold), skipping throttle update for broker 1041
```

### CODE REVIEW

File: cmd/autothrottle/throttles.go [hunk 1]
```diff
@@ -130,92 +149,67 @@ func updateReplicationThrottle(params *ReplicationThrottleConfigs) error {
...
-	Log success.
-	***********/
-
-	// Write event.
-	var b bytes.Buffer
-	b.WriteString(fmt.Sprintf("Replication throttle of %0.2fMB/s set on the following brokers: %v\n",
-		replicationCapacity, allBrokers))
-	b.WriteString(fmt.Sprintf("Topics currently undergoing replication: %v", params.topics))
-	params.events.Write("Broker replication throttle set", b.String())
+	// Write event. TODO(jamie): update for per-broker rates.
```
Comment (jamiealquiza) [inline cmd/autothrottle/throttles.go]: (Do)

File: cmd/autothrottle/throttles.go [hunk 2]
```diff
@@ -279,72 +273,153 @@ func getReassigningBrokers(r kafkazk.Reassignments, zk kafkazk.Handler) (reassig
...
 		}
-	}
 
-	// Destination brokers.
-	for b := range reassigning.dst {
-		if broker, exists := bm[b]; exists {
-			throttledBrokers.Dst = append(throttledBrokers.Dst, broker)
-		} else {
-			return 0.00, 0.00, event, fmt.Errorf("Broker %d not found in broker metrics", b)
+		// Source (outbound) throttle.
```
Comment (jamiealquiza) [inline cmd/autothrottle/throttles.go]: (cleanup)
